sympy/physics/mechanics/_pathway.py

```diff
@@ -0,0 +1,273 @@
+        return self._true_length()
+
+    @property
+    def shortening_velocity(self) -> Expr:
+        """An expression representing the pathway's shortening velocity."""
+        return self._true_shortening_velocity()
+
+    def __repr__(self) -> str:
+        return (
+            f'{self.__class__.__name__}(origin={self.origin}, '
+            f'insertion={self.insertion})'
+        )
+
+
+class LinearPathway(PathwayBase):
+    """Linear pathway between origin and insertion.
+
+    Explanation
+    ===========
+
+    A linear pathway forms a straight-line segment between two points and is
+    the simplest pathway that can be formed. It will not interact with any
+    other objects in the system, i.e. a ``LinearPathway`` will interest other
+    objects to ensure that the path between its two ends (its origin and
+    insertion) is the shortest possible.
+
+    Examples
+    ========
+
+    As the ``_pathway.py`` module is experimental, it is not yet part of the
+    ``sympy.physics.mechanics`` namespace and must be imported directly from
+    the ``sympy.physics.mechanics._pathway`` module.
+
+    >>> from sympy.physics.mechanics._pathway import LinearPathway
+
+    To construct a pathway, two points are required to be passed to the
+    ``origin`` and ``insertion`` parameters.
+
+    >>> from sympy.physics.mechanics import Point
+    >>> origin = Point('pO')
+    >>> insertion = Point('pI')
+    >>> linear_pathway = LinearPathway(origin=origin, insertion=insertion)
+    >>> linear_pathway
+    LinearPathway(origin=pO, insertion=pI)
+
+    The pathway create above isn't very interesting without the positions and
+    velocities of its origin and insertion points being described. Without this
+    its not possible to describe how the pathway moves, i.e. its length or its
+    shortening velocity.
+
+    >>> from sympy.physics.mechanics import ReferenceFrame
+    >>> from sympy.physics.vector import dynamicsymbols
+    >>> N = ReferenceFrame('N')
+    >>> q = dynamicsymbols('q')
+    >>> insertion.set_pos(origin, q * N.x)
+    >>> insertion.pos_from(origin)
+    q(t)*N.x
+
+    A pathway's length can be accessed via its ``length`` attribute.
+
+    >>> linear_pathway.length
+    sqrt(q(t)**2)
+
+    Note how what appears to be an overly-complex expression is returned. This
+    is actually required as it ensures that a pathway's length is always
+    positive.
+
+    A pathway's shortening velocity can be accessed similarly via its
+    ``shortening_velocity`` attribute.
+
+    >>> linear_pathway.shortening_velocity
+    -q(t)*Derivative(q(t), t)/sqrt(q(t)**2)
+
+    Parameters
+    ==========
+
+    origin : Point
+        The ``Point`` from which the pathway originates.
+    insertion : Point
+        The ``Point`` into which the pathway inserts.
+
+    """
+
+    def __init__(
+        self,
+        *,
+        origin: Point,
+        insertion: Point,
+    ) -> None:
+        """Initializer for ``LinearPathway``.
```

Comment:
Is this common in `sympy` to also have a docstring in the `__init__`?

Reply:
Not common nor required. I however think it's good practice. When calling `help` on the class within a REPL session the constructor (`__init__` method) is shown. If the `__init__` has parameters then I think it's useful as this is the only place where the call signature for instantiation is documented. Yes, it does mean a lot of repetition, but it is also the most complete way to do it.

Reply:
Another option would be using a regular expression to generate these, would be fun, though overkill.

Reply:
I've tried this kind of thing before but found that it just makes the source code less readable and can make it impenetrable to people unfamiliar with the codebase. Sometimes it's okay to repeat yourself 🙂